src/reflect/scala/reflect/internal/pickling/UnPickler.scala

```diff
@@ -721,7 +721,7 @@ abstract class UnPickler {
       private[this] val p = phase
       protected def completeInternal(sym: Symbol) : Unit = try {
         completingStack += sym
-        val tp = at(i, () => readType(sym.isTerm)) // after NMT_TRANSITION, revert `() => readType(sym.isTerm)` to `readType`
+        val tp = at(i, () => readType()) // after NMT_TRANSITION, revert `() => readType()` to `readType`
```

Comment:
I think you've done the TODO.

Also, you could save a few cycles by removing the temporary fix below.

Reply:
I was asking earlier today if anyone knew what that TODO meant. https://gitter.im/scala/contributors?at=5dbdcf609825bd6bac14ff6e

Reply:
As to the temporary fix, it seems @lrytz was the author, so he may advise on the matter (or remove it himself).

Reply:
NMT is `NullaryMethodType`, which didn't always exist, apparently.

Reply:
Indeed (f0bff86d31). The commend can be removed, `at(i, readType)`.

The fix below can also go.

Reply:
By "the fix below", are you referring to lines 726 to 738, and that `fixLocalChildTp` can simply be replaced with `tp`?

Reply:
Yes, remove `fixLocalChildTp` and use `tp` instead.